lib/headRoute.js

```diff
@@ -0,0 +1,30 @@
+function headRouteOnSendHandler (req, reply, payload, done) {
+  // If payload is undefined
+  if (payload === undefined) {
+    reply.header('content-length', '0')
+    return done(null, null)
+  }
+
+  // If is a stream, we remove content-type header without appending
+  // content-length one, and resume it right away
+  if (typeof payload.resume === 'function') {
+    reply.removeHeader('content-type')
+    payload.resume()
```

Comment:
You'll need to add an `on('error')` handler. Also add a test for this case.

Reply:
Curious, should throw in case of an error or just use the logger to show it up?

Reply:
You need to call the callback `done(err)`.

Reply:
Sure! done in 08f3772

Reply:
Now that I look at it is a bit more complex. Sorry about the wrong turn - you should not call the callback because it would make it call the callback twice which is not really something we want.

You should just log the error - the response was already sent, so there is nothing else to do.

Reply:
No worries, I think that also explains why was quite hard to test. Turning it into a log output then